base/server/src/main/java/org/dogtagpki/server/cli/SubsystemDBReplicationSetupCLI.java

```diff
@@ -0,0 +1,292 @@
+import com.netscape.cmscore.ldapconn.LdapBoundConnFactory;
+import com.netscape.cmscore.ldapconn.LdapBoundConnection;
+import com.netscape.cmscore.ldapconn.LdapConnInfo;
+import com.netscape.cmscore.ldapconn.PKISocketConfig;
+import com.netscape.cmscore.ldapconn.PKISocketFactory;
+import com.netscape.cmsutil.password.IPasswordStore;
+import com.netscape.cmsutil.password.PasswordStoreConfig;
+
+import netscape.ldap.LDAPConnection;
+
+/**
+ * @author Endi S. Dewata
+ */
+public class SubsystemDBReplicationSetupCLI extends SubsystemCLI {
+
+    public static Logger logger = LoggerFactory.getLogger(SubsystemDBReplicationSetupCLI.class);
+
+    public SubsystemDBReplicationSetupCLI(CLI parent) {
+        super("setup", "Set up " + parent.parent.parent.getName().toUpperCase() + " database replication", parent);
+    }
+
+    @Override
+    public void createOptions() {
+
+        options.addOption(null, "master-replication-port", true, "Master replication port");
+        options.addOption(null, "replica-replication-port", true, "Replica replication port");
+        options.addOption(null, "replication-security", true, "Replication security: SSL, TLS, None");
+
+        options.addOption("v", "verbose", false, "Run in verbose mode.");
+        options.addOption(null, "debug", false, "Run in debug mode.");
+        options.addOption(null, "help", false, "Show help message.");
+    }
+
+    @Override
+    public void execute(CommandLine cmd) throws Exception {
+
+        if (cmd.hasOption("debug")) {
+            PKILogger.setLevel(Level.DEBUG);
+
+        } else if (cmd.hasOption("verbose")) {
+            PKILogger.setLevel(Level.INFO);
+        }
+
+        String masterReplicationPort = cmd.getOptionValue("master-replication-port");
+        String replicaReplicationPort = cmd.getOptionValue("replica-replication-port");
+        String replicationSecurity = cmd.getOptionValue("replication-security");
+
+        initializeTomcatJSS();
+        String subsystem = parent.parent.parent.getName();
+        EngineConfig cs = getEngineConfig(subsystem);
+        cs.load();
+
+        LDAPConfig ldapConfig = cs.getInternalDBConfig();
+        String database = ldapConfig.getDatabase();
+        String baseDN = ldapConfig.getBaseDN();
+
+        String instanceID = cs.getInstanceID();
+
+        PasswordStoreConfig psc = cs.getPasswordStoreConfig();
+        IPasswordStore passwordStore = IPasswordStore.create(psc);
+
+        LDAPConnectionConfig connConfig = ldapConfig.getConnectionConfig();
+        LDAPAuthenticationConfig authConfig = ldapConfig.getAuthenticationConfig();
+
+        LdapConnInfo connInfo = new LdapConnInfo(connConfig);
+
+        LdapAuthInfo authInfo = new LdapAuthInfo();
+        authInfo.setPasswordStore(passwordStore);
+        authInfo.init(
+                authConfig,
+                connInfo.getHost(),
+                connInfo.getPort(),
+                connInfo.getSecure());
```

Comment:
I refactored this part, you must have done this change before it was merged. You can get `authInfo` from the superclass now:
`LdapAuthInfo authInfo = getAuthInfo(passwordStore, connInfo, ldapConfig);`